Utils/Extensions/StringExtensions.swift

```diff
@@ -40,4 +44,105 @@ extension String {
     public var asURL: NSURL? {
         return NSURL(string: self)
     }
+
+    private func shaw256(salt: String, payload: String) -> String? {
+        var saltData = salt.dataUsingEncoding(NSUTF8StringEncoding)
+        var payloadData = payload.dataUsingEncoding(NSUTF8StringEncoding)
+        let hash = NSMutableData(length: Int(CC_SHA256_DIGEST_LENGTH))
+
+        let operation: CCOperation = UInt32(kCCHmacAlgSHA256)
+        CCHmac(operation, saltData!.bytes, saltData!.length, payloadData!.bytes, payloadData!.length, hash!.mutableBytes);
```

Comment:
All these exclamation marks. This will crash if any of the above operations fail and return nil. Needs to go into an `if let` to not crash. Since sha256 returns an optional there is no reason to do this. Check to be safe and return `nil` in case of failures.